Add-on v1.2.0 documentation (#747)
nrfconnect/sdk-sidewalk: doc/releases_and_migration/migration_guide_addon_v120.rst

.. _migration_guide_addon_v120:

Migration guide for Amazon Sidewalk Add-On v1.2.0
##################################################

.. contents::
   :local:
   :depth: 2

This guide assists you in migrating from Amazon Sidewalk Add-On v1.1.0 to v1.2.0, which includes the upgrade to NCS v3.3.0 and the transition from Partition Manager to devicetree-based flash partitioning.

Overview
********

> Amazon Sidewalk Add-On v1.2.0 introduces support for two new hardware platforms (nRF54LV10 and nRF54LM20) and aligns the flash partition layout with the NCS v3.3.0 transition away from Partition Manager.

Reviewer: ```suggestion
Amazon Sidewalk Add-On v1.2.0 introduces support for three new hardware platforms (nRF54LV10A,  nRF54LM20A and nRF54LM20B) and aligns the flash partition layout with the NCS v3.3.0 transition away from Partition Manager.
```

Key updates in this release include:

> * Including new board support for the nRF54LV10 and nRF54LM20 SoCs in Bluetooth LE-only samples.

Reviewer: ```suggestion
* Including new board support for the nRF54LV10A, nRF54LM20A and nRF45LM20B SoCs in Bluetooth LE-only samples.
```

* Updating the nRF Connect SDK from v3.0.0 to v3.3.0.
* Migration of flash layout from :file:`pm_static*.yml` files to devicetree overlays for all supported boards except nrf5340 and non-secure variants (see :ref:`migration_120_partition_manager`).

> * On the nRF52 Series, new `Trusted storage`_ AEAD key backend default (see :ref:`migration_120_trusted_storage_nrf52`).

Reviewer: ```suggestion
* On the nRF52 Series, changed the default AEAD key of `Trusted storage`_ (see :ref:`migration_120_trusted_storage_nrf52`).
```

* On the nRF54L10 SoC, corrected application-core RRAM size and relocated the manufacturing storage partition (see :ref:`migration_120_mfg_storage_nrf54l10`).

Migration steps
***************

#. Navigate to the Amazon Sidewalk Add-on repository (the :file:`sidewalk` directory in your west workspace).

#. Update the Add-on repository to the latest ``main`` branch from the `sdk-sidewalk`_ GitHub repository.

>    The commands below assume a Git remote named ``ncs`` that points to `sdk-sidewalk`_.

Reviewer 2: nit: perhaps `origin` is more probable remote name if a user uses VSCode or west for initializing the add-on.

Author: I followed the NCS doc naming convention for git remotes:
"origin usually points to the user’s personal copy of the repository.
ncs is used to point to the nRF Connect SDK repository."
~ https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/dev_model_and_contributions/managing_code.html#forking_a_repository_of_the_nrf_connect_sdk

   If your remote uses a different name, substitute it for ``ncs``.

   .. code-block:: console

      git fetch ncs
      git checkout main

>       git pull ncs main

Reviewer 2: If you do `pull`, the previous `fetch` is unnecessary.

Reviewer 2: Actually, do we want a user to update to `main` and not `v1.2.0`?

If you maintain a custom board with a :file:`pm_static*.yml` file, see the `Migrating partition configuration from Partition Manager to devicetree (DTS)`_ page in the NCS documentation.

To confirm that migration succeeded, run a pristine build on your board:

   .. code-block:: console

      west build -p -b <your_board> <sidewalk_application>

.. _migration_120_trusted_storage_nrf52:

Trusted storage on the nRF52 Series
***********************************

With NCS v3.3.0, Amazon Sidewalk Add-on v1.2.0 changes the default `Trusted storage`_ AEAD key backend on the nRF52 Series from the ``CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_DERIVE_FROM_HUK`` to ``CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_HASH_UID`` Kconfig option.
This applies when Sidewalk PSA key storage is enabled (``CONFIG_SIDEWALK_CRYPTO_PSA_KEY_STORAGE``).

Choose the backend based on whether devices already store Sidewalk keys in trusted storage:

* Field firmware updates - Keep the ``CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_DERIVE_FROM_HUK`` Kconfig option enabled so that existing trusted storage entries remain readable.

>   The HUK partition contents do not matter for this backend unless you use the nRF Secure Immutable Bootloader (NSIB).

Reviewer 2: ```suggestion
  The HUK partition contents do not matter for this backend unless you use the `nRF Secure Immutable Bootloader (NSIB)`_.
```

.. _migration_guide_addon_v120:

Migration guide for Amazon Sidewalk Add-On v1.2.0
##################################################

.. contents::
   :local:
   :depth: 2

This guide assists you in migrating from Amazon Sidewalk Add-On v1.1.0 to v1.2.0, which includes the upgrade to NCS v3.3.0 and the transition from Partition Manager to devicetree-based flash partitioning.

Overview
********

Amazon Sidewalk Add-On v1.2.0 introduces support for two new hardware platforms (nRF54LV10 and nRF54LM20) and aligns the flash partition layout with the NCS v3.3.0 transition away from Partition Manager.

Key updates in this release include:

* Including new board support for the nRF54LV10 and nRF54LM20 SoCs in Bluetooth LE-only samples.

> * Migration of flash layout from :file:`pm_static*.yml` files to devicetree overlays for supported boards.

Reviewer: (except nRF5340 and nrf54l15 ns variant)?

* Updating the nRF Connect SDK from v3.0.0 to v3.3.0.

Migration steps
***************

#. Update your west configuration to use the latest Add-on version:

   .. code-block:: console

>       west update

Reviewer 2: not a big deal, but this is already a part of add-on installation, right?

Author: yes, but here it is to update after nrf revision update in sidewalk west.yaml.
I split the ncs update part form parition manager for clarity

#. Perform a clean build:

   .. code-block:: console

      west build -p -b <board_target> <your_application>

> #. If you maintain a custom board with a :file:`pm_static*.yml` file, see the `Migrating partition configuration from Partition Manager to devicetree (DTS)`_ page in the NCS documentation.

Reviewer 2: "Partition Manager has been deprecated starting from nRF Connect SDK 3.3.0. If your application uses Partition Manager configuration files (e.g :file:`pm_static*.yml` files) to define partition layout...".
To give a bit more context. Also, this is not strictly related to a custom board; rather a custom application.

Backward compatibility
**********************

This release preserves backward compatibility for the application-facing Sidewalk API, with the following exceptions:

* nRF54L10 flash layout — Updated the manufacturing HEX file location.
  Custom board configurations must update their DTS overlays accordingly.

> * Bluetooth LE PAL connection symbols has been renamed in :file:`subsys/sal/sid_pal/include/sid_ble_connection.h``.

Reviewer 2: ```suggestion
* Bluetooth LE PAL connection symbols have been renamed in :file:`subsys/sal/sid_pal/include/sid_ble_connection.h``.
```

Key updates in this release include:

* Including new board support for the nRF54LV10 and nRF54LM20 SoCs in Bluetooth LE-only samples.
* Updating the nRF Connect SDK from v3.0.0 to v3.3.0.
* Migration of flash layout from :file:`pm_static*.yml` files to devicetree overlays for all supported boards except nrf5340 and non-secure variants (see :ref:`migration_120_partition_manager`).
* On the nRF52 Series, new `Trusted storage`_ AEAD key backend default (see :ref:`migration_120_trusted_storage_nrf52`).

> * On the nRF54L10 SoC, corrected application-core RRAM size and relocated the manufacturing storage partition (see :ref:`migration_120_mfg_storage_nrf54l10`).

Reviewer 3: Should we just redirect to the release notes file?

Author: I think migration intruction has value in this case, to aware customers about nrf54l10 parition address changes (together with pm yaml -> dts)

      west build -p -b <your_board> <sidewalk_application>

.. _migration_120_trusted_storage_nrf52:

Trusted storage on nRF52

> ************************

Reviewer 3: ```suggestion
Trusted storage on the nRF52 Series
***********************************

> With NCS v3.3.0, Amazon Sidewalk Add-on v1.2.0 changes the default `Trusted storage`_ AEAD key backend on the nRF52 Series from ``CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_DERIVE_FROM_HUK`` to ``CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_HASH_UID``.

Reviewer 3: ```suggestion
With NCS v3.3.0, Amazon Sidewalk Add-on v1.2.0 changes the default `Trusted storage`_ AEAD key backend on the nRF52 Series from the ``CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_DERIVE_FROM_HUK`` to ``CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_HASH_UID`` Kconfig option.
```

This applies when Sidewalk PSA key storage is enabled (``CONFIG_SIDEWALK_CRYPTO_PSA_KEY_STORAGE``).

Choose the backend based on whether devices already store Sidewalk keys in trusted storage:

> * **Field firmware updates** — Keep ``CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_DERIVE_FROM_HUK`` so existing trusted-storage entries remain readable.

Reviewer 3: ```suggestion
* Field firmware updates - Keep the ``CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_DERIVE_FROM_HUK`` Kconfig option enabled so that existing trusted storage entries remain readable.

  The HUK partition contents do not matter for this backend unless you use the nRF Secure Immutable Bootloader (NSIB).

> * **New products and factory-fresh devices** — You may use the new default ``CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_HASH_UID``, or explicitly select ``CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_DERIVE_FROM_HUK``.

Reviewer 3: ```suggestion
* New products and factory-fresh devices - You can either keep the new default (``CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_HASH_UID``) or explicitly select the ``CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_DERIVE_FROM_HUK`` Kconfig option.

  The HUK-derived option can offer stronger key binding, but requires NSIB to load the HUK partition into the CryptoCell and adds a bootloader stage.

> For background on Sidewalk key storage, see the Secure Key Storage section in :ref:`sidewalk_end_device` and the `Trusted storage`_ documentation in the NCS.

Reviewer 3: ```suggestion
For background on Sidewalk key storage, see the Secure Key Storage section in :ref:`sidewalk_end_device` and the `Trusted storage`_ documentation in the nRF Connect SDK.

.. warning::

>    After Sidewalk keys are written to trusted storage, changing ``CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_*`` or other trusted-storage crypto settings makes existing entries unreadable and can brick in-field devices.

Reviewer 3: ```suggestion
   After Sidewalk keys are written to trusted storage, changing the ``CONFIG_TRUSTED_STORAGE_BACKEND_AEAD_KEY_*`` Kconfig option or other trusted-storage crypto settings makes existing entries unreadable and can brick in-field devices.
```

.. _migration_120_mfg_storage_nrf54l10:

Manufacturing storage on nRF54L10

> *********************************

Reviewer 3: ```suggestion
Manufacturing storage on the nRF54L10 device
********************************************
```

> Amazon Sidewalk Add-on v1.2.0 corrects the nRF54L10 application-core RRAM map and moves Sidewalk manufacturing data out of Partition Manager into devicetree.

Reviewer 3: ```suggestion
Amazon Sidewalk Add-on v1.2.0 corrects the application-core RRAM map on the nRF54L10 SoC and moves Sidewalk manufacturing data out of Partition Manager into devicetree.
```

In Add-on v1.1.0, Partition Manager placed the ``mfg_storage`` region at ``0xFF000`` (4 KB), based on a 1024 KB RRAM assumption.
The SoC exposes 1012 KB of application-core RRAM, so that address was past the end of physical memory.
The in-tree DTS overlay now defines ``mfg_storage`` at ``0xFC000`` (4 KB), at the end of the 1012 KB RRAM region.
The PAL reads this partition through ``FIXED_PARTITION_OFFSET(mfg_storage)`` when Partition Manager is disabled.

> If you develop for nRF54L10, complete the following steps:

Reviewer 3: Since the section is for L10 specifically
```suggestion
Complete the following steps:
```
